### PR TITLE
update roadmap for 2024

### PR DESCRIPTION
**Pull request by HardenedSteel**
- Remove return address proposal which had no activity/development.
- Mark Haveno as completed.
- Remove ring size task which is duplicate of Seraphis.
- Add "proposed task" status.
  - Add ideas from Monero Research Lab

### CODE REVIEW

**Comment by HardenedSteel** [comment]
^ updated the commit description

https://deploy-preview-2310--barolo-time-757cf9.netlify.app/resources/roadmap/
LGTM

**Comment by plowsof** [comment]
Thanks! Some initial thoughts:
- We could link the open mrl open research issue as the only proposed task https://github.com/monero-project/research-lab/issues/94 (Rucknium is in MRL and we can let them maintain/update that list appropriately) 
- FCMP++ can be listed (linking to kayabanerves recent blog) as on-going. https://www.getmonero.org/2024/04/27/fcmps.html
- eth-xmr swaps are "completed" but the last update in their matrix room was that not many people came forward to test. But this is my unqualified opinion. I think bch-xmr should be listed as also on-going. Last i seen they where working on a monero web wallet. Depends how we define completed : a swap on main/testnet or a working gui/cli that has peers to swap.with which we can link users to.
- we also seen an MVP xmr-ltc testnet atomic swap from kayabanerve @ https://bounties.monero.social/posts/112/21-050m-xmr-to-ltc-atomic-swap but  im not sure of any projects integrating/actively working on it  (early days)
- monerod rust implementation - Cuprate

**Comment by nahuhh** [comment]
> we also seen an MVP xmr-ltc testnet atomic swap from kayabanerve @ https://bounties.monero.social/posts/112/21-050m-xmr-to-ltc-atomic-swap but im not sure of any projects integrating/actively working on it (early days)

xmr <> ltc has been working via basicswap for a long time. Currently uses adaptor sigs

**Comment by HardenedSteel** [comment]
tbh XMR-BCH doesn't look very completed to me. 

and work on Seraphis not started should i change it to planned?

**Comment by HardenedSteel** [comment]
@nahuhh since when? for adding to roadmap

**Comment by nahuhh** [comment]
> @nahuhh since when? for adding to roadmap

Just saw this.
ltc<>xmr has been live on basicswap since dec 2022

**Comment by plowsof** [comment]
We can get some MRL input on removing tych. That open ccs was refunded to the GF a while ago

**Comment by nahuhh** [comment]
Triptych should be removed. Its not ongoing nor upcoming. Iirc it evolved into lelantus spark and seraphis

File: _i18n/en.yml [hunk 1]
```diff
@@ -820,27 +821,31 @@ roadmap:
   clsag: Concise linkable spontaneous anonymous group (CLSAG) signatures
   tryptych: "Triptych: logarithmic-sized linkable ring signatures with applications"
```
Comment (nahuhh) [inline _i18n/en.yml:822]: Remove

File: resources/roadmap/index.md [hunk 2]
```diff
@@ -214,23 +214,34 @@ meta_descr: meta_descr.roadmap
...
                                     <li class="ongoing"><a href="https://github.com/Rucknium/OSPEAD">{% t roadmap.ospead %}</a></li>
-                                    <li class="ongoing">{% t roadmap.ringsize-sera-tryp %}</li>
-                                    <li class="ongoing">{% t roadmap.tryptych %}</li>
                                     <li class="ongoing">{% t roadmap.xmr-eth-atomicswaps %}</li>
                                     <li class="ongoing">{% t roadmap.bp++ %}</li>
                                     <li class="ongoing">{% t roadmap.lws-subaddress %}</li>
-                                    <li class="upcoming">{% t roadmap.returnaddr %}</li>
-                                    <li class="upcoming">{% t roadmap.layer2 %}</li>
+                                    <li class="ongoing">{% t roadmap.cuprate %}</li>
+                                    <li class="upcoming">{% t roadmap.tryptych %}</li>
```
Comment (nahuhh) [inline resources/roadmap/index.md:241]: Remove

**Review by nahuhh — CHANGES_REQUESTED** [review]
Triptych isn't coming. My understanding is that it has evolved into Lelantus Spark and Seraphis

File: resources/roadmap/index.md (hunk 2 above)
Comment (plowsof) [reply resources/roadmap/index.md:241]: seems sane. using this as a reference: see 'proposal closed' https://ccs.getmonero.org/proposals/cypherstack-sarang-triptych-research.html

File: resources/roadmap/index.md [hunk 3]
```diff
@@ -214,23 +214,34 @@ meta_descr: meta_descr.roadmap
...
+                    <label for="tab-6" tabindex="0"></label>
+                    <input id="tab-6" type="radio" name="tabs" aria-hidden="true" checked>
+                    <h2>2024</h2>
+                    <div class="tabPanel-content info-block">
+                        <h2>2024</h2>
+                            <ul>
+                                <h3 class="months">{% t roadmap.jan %}</h3>
+                                    <li class="completed">{% t roadmap.xmrbch-atomicswap %}</li>
+                                <h3 class="months">{% t roadmap.may %}</h3>
+                                    <li class="completed">{% t roadmap.haveno-mainnet %}</li>
```
Comment (plowsof) [inline resources/roadmap/index.md:226]:
```suggestion
                                    <li class="completed"><a href="https://haveno.exchange">{% t roadmap.haveno-mainnet %}</a></li>
```

File: README.md [hunk 4]
```diff
@@ -293,6 +293,7 @@ Where `STATUS` can be:
 - `completed` for completed milestones
 - `ongoing` for milestones currently being worked on
 - `upcoming` for milestones not yet completed or in progress
+- `proposed` for milestones are open to research and consideration
```
Comment (plowsof) [inline README.md:296]:
```suggestion
- `proposed` for milestones open to research and consideration
```

File: _i18n/en.yml [hunk 5]
```diff
@@ -818,29 +819,32 @@ roadmap:
...
   released-0-17: CLI and GUI 0.17 'Oxygen Orion' released
   
-  returnaddr: Return addresses
   atomicswaps: Monero <-> Bitcoin atomic swaps
   bplus: Bulletproofs+
   p2pool: P2Pool released
   haveno: Haveno
 
-  ringsize-sera-tryp: increase ring size to >100
   seraphis-jamtis: Seraphis & Jamtis
```
Comment (plowsof) [inline _i18n/en.yml:831]:
```suggestion
  seraphis-jamtis: Seraphis codebase & Jamtis
```
this small but important change will reflect the projects change in direction. i can than ask rbrunner to clarify this at the top of the wiki we link to. this comes from some comment kayabanerve left in #monero-research-lounge. ill post logs below

**Comment by plowsof** [comment]
edited excerpts without full context from an informal chat in monero lounge regarding layer2's below: (which is being deleted in this PR)
```
<kayabanerve> [...] FCMP++s enable some PCNs as Seraphis (the protocol) would.
<k​ayabanerve> So we are indirectly working on those, even if I don't support the enabled instances I know of.
```
Rucknium mentions cores intentions of making layer 1 scale and "Since 7 years ago, a few papers on Monero payment channels have been published. No production code implementing them AFAIK"

My thoughts on layer2's are that this was probably added because it's a popular buzzword and people bring it up a lot

**Comment by nahuhh** [comment]
LWS subaddress support was completed as of Dec 2023

**Comment by HardenedSteel** [comment]
@nahuhh  do you have link for it?

**Comment by nahuhh** [comment]
https://github.com/vtnerd/monero-lws/commit/b4426b4a74f1ab83f076288c34816447bc1577ba

I spoke with vtnerd to confirm subaddress support

**Comment by HardenedSteel** [comment]
didn't add the commit link, not very useful for the roadmap

**Comment by nahuhh** [comment]
The ETH<>XMR atomic swaps should likely move to completed as well. I'm not sure the date. I think May 2023, but need to double check

File: resources/roadmap/index.md [hunk 6]
```diff
@@ -221,16 +235,13 @@ meta_descr: meta_descr.roadmap
...
                                     <li class="ongoing"><a href="https://github.com/Rucknium/OSPEAD">{% t roadmap.ospead %}</a></li>
-                                    <li class="ongoing">{% t roadmap.ringsize-sera-tryp %}</li>
-                                    <li class="ongoing">{% t roadmap.tryptych %}</li>
                                     <li class="ongoing">{% t roadmap.xmr-eth-atomicswaps %}</li>
                                     <li class="ongoing">{% t roadmap.bp++ %}</li>
-                                    <li class="ongoing">{% t roadmap.lws-subaddress %}</li>
-                                    <li class="upcoming">{% t roadmap.returnaddr %}</li>
-                                    <li class="upcoming">{% t roadmap.layer2 %}</li>
+                                    <li class="ongoing">{% t roadmap.cuprate %}</li>
+                                    <li class="upcoming"><a href="https://github.com/seraphis-migration/strategy/wiki">{% t roadmap.seraphis-jamtis %}</a></li>
```
Comment (plowsof) [inline resources/roadmap/index.md:242]:
```suggestion
                                    <li class="upcoming"><a href="https://github.com/seraphis-migration/strategy/wiki/Our-Wallet-Rewrite-Project-After-Priority-Switched-From-Seraphis-to-FCMPs">{% t roadmap.seraphis-jamtis %}</a></li>
```
rbrunner has added a new entry to the seraphis wiki above. it explains the "Seraphis & Jamtis" -> "Seraphis codebase & Jamtis" shift.

Comment (HardenedSteel) [reply resources/roadmap/index.md:242]: I believe we shouldn't link the FCMP for seraphis roadmap, instead we should link for FCMP roadmap itself

File: resources/roadmap/index.md [hunk 7]
```diff
@@ -221,16 +235,13 @@ meta_descr: meta_descr.roadmap
...
+                                    <li class="upcoming"><a href="https://www.getmonero.org/2024/04/27/fcmps.html">{% t roadmap.fcmp-plus %}</a></li>
```
Comment (HardenedSteel) [inline resources/roadmap/index.md:243]: this one

Comment (plowsof) [reply resources/roadmap/index.md:243]: Can you show me what you want to change/remove/keep? its not clear to me

Comment (HardenedSteel) [reply resources/roadmap/index.md:243]: move ``<a href="https://github.com/seraphis-migration/strategy/wiki">`` line 243 from line 242

Comment (plowsof) [reply resources/roadmap/index.md:243]: make these changes as @nahuhh seems to agree and we can review further. What will `Seraphis codebase and jamtis` link to then?

Comment (nahuhh) [inline resources/roadmap/index.md:243]:
```suggestion
                                    <li class="upcoming"><a href="https://github.com/seraphis-migration/strategy/wiki/Our-Wallet-Rewrite-Project-After-Priority-Switched-From-Seraphis-to-FCMPs">{% t roadmap.seraphis-jamtis %}</a></li>
```

@HardenedSteel like this, right?

Comment (HardenedSteel) [reply resources/roadmap/index.md:243]: yes correct like this one @plowsof

Comment (plowsof) [reply resources/roadmap/index.md:243]: this + removal of the hyperlink on seraphis codebase+jamtis LGTM

**Comment by nahuhh** [comment]
bumping this:
Went from prerelease to release in may 2023

Tldr: BCH is marked as completed, so eth should be as well.

> The ETH<>XMR atomic swaps should likely move to completed as well.

plowsof:
>  Depends how we define completed : a swap on main/testnet

this is how id define completed ^.  liquidity or consumer grade tooling isn't essential to completion. 

> I'm personally ok with listing both BCH and ETH as completed. I don't believe we waited for [unstoppableswaps] before considering BTC<>XMR as complete, but considered "complete" to imply that the protocol was functional

File: resources/roadmap/index.md [hunk 8]
```diff
@@ -192,26 +190,43 @@ meta_descr: meta_descr.roadmap
...
+                    <label for="tab-6" tabindex="0"></label>
+                    <input id="tab-6" type="radio" name="tabs" aria-hidden="true" checked>
+                    <h2>2024</h2>
+                    <div class="tabPanel-content info-block">
+                        <h2>2024</h2>
+                            <ul>
+                                <h3 class="months">{% t roadmap.jan %}</h3>
+                                    <li class="completed">{% t roadmap.xmrbch-atomicswap %}</li>
+                                <h3 class="months">{% t roadmap.may %}</h3>
+                                    <li class="completed">{% t roadmap.xmr-eth-atomicswaps %}</li>
```
Comment (nahuhh) [inline resources/roadmap/index.md:228]: May 2023

**Review by nahuhh — APPROVED** [review]

**Comment by HardenedSteel** [comment]
finally

**Review by plowsof — APPROVED** [review]
